tests/component_sklearn/test_tree_models.py

```diff
@@ -0,0 +1,154 @@
+        ],
+    }
+    expected_transaction_name = "test_tree_models:_test"
+    if six.PY3:
+        expected_transaction_name = "test_tree_models:test_model_methods_wrapped_in_function_trace.<locals>._test"
+
+    @validate_transaction_metrics(
+        expected_transaction_name,
+        scoped_metrics=expected_scoped_metrics[tree_model_name],
+        rollup_metrics=expected_scoped_metrics[tree_model_name],
+        background_task=True,
+    )
+    @background_task()
+    def _test():
+        run_tree_model()
+
+    _test()
+
+
+def test_multiple_calls_to_model_methods(tree_model_name, run_tree_model):
+    # Note: in the following expected metrics, predict and predict_proba are called by
+    # score and predict_log_proba so they are expected to be called twice as often as
+    # the other methods.
+    expected_scoped_metrics = {
+        "ExtraTreeRegressor": [
+            ("MLModel/Sklearn/Named/ExtraTreeRegressor.fit", 1),
+            ("MLModel/Sklearn/Named/ExtraTreeRegressor.predict", 4),
+            ("MLModel/Sklearn/Named/ExtraTreeRegressor.score", 2),
+        ],
+        "DecisionTreeClassifier": [
+            ("MLModel/Sklearn/Named/DecisionTreeClassifier.fit", 1),
+            ("MLModel/Sklearn/Named/DecisionTreeClassifier.predict", 4),
+            ("MLModel/Sklearn/Named/DecisionTreeClassifier.score", 2),
+            ("MLModel/Sklearn/Named/DecisionTreeClassifier.predict_log_proba", 2),
+            ("MLModel/Sklearn/Named/DecisionTreeClassifier.predict_proba", 4),
+        ],
+        "ExtraTreeClassifier": [
+            ("MLModel/Sklearn/Named/ExtraTreeClassifier.fit", 1),
+            ("MLModel/Sklearn/Named/ExtraTreeClassifier.predict", 4),
+            ("MLModel/Sklearn/Named/ExtraTreeClassifier.score", 2),
+            ("MLModel/Sklearn/Named/ExtraTreeClassifier.predict_log_proba", 2),
+            ("MLModel/Sklearn/Named/ExtraTreeClassifier.predict_proba", 4),
+        ],
+        "DecisionTreeRegressor": [
+            ("MLModel/Sklearn/Named/DecisionTreeRegressor.fit", 1),
+            ("MLModel/Sklearn/Named/DecisionTreeRegressor.predict", 4),
+            ("MLModel/Sklearn/Named/DecisionTreeRegressor.score", 2),
+        ],
+    }
+    expected_transaction_name = "test_tree_models:_test"
+    if six.PY3:
+        expected_transaction_name = "test_tree_models:test_multiple_calls_to_model_methods.<locals>._test"
+
+    @validate_transaction_metrics(
+        expected_transaction_name,
+        scoped_metrics=expected_scoped_metrics[tree_model_name],
+        rollup_metrics=expected_scoped_metrics[tree_model_name],
+        background_task=True,
+    )
+    @background_task()
+    def _test():
+        x_test = [[2.0, 2.0], [2.0, 1.0]]
+        y_test = [1, 1]
+
+        model = run_tree_model()
+
+        model.predict(x_test)
+        model.score(x_test, y_test)
+        # Only classifier models have proba methods.
+        if tree_model_name in ("DecisionTreeClassifier", "ExtraTreeClassifier"):
```

Comment:
Replacing with a `hasattr` might make this more portable so it can be reused for other tests.

Reply:
I suspect this is going to be very specific to the types of models so I'm gonna leave it for now as is. If it turns out to be useful as we evaluate more models we can totally do this though!

Reply:
Good thinking!